src/constant_value.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use rustc::hir::def_id::DefId;
```

Comment:
What is this `DefId` thing? Is a `DefId` globally unique?

Reply:
It is short for Definition Identifier and it is a handle that you can use to queries in a single compiler session. It is not globally unique. Things may change on this front, apparently.

The information retrieved from queries can be computed on demand, so that the compiler does not hang on to too much memory.